openedx/core/djangoapps/credit/services.py

```diff
@@ -0,0 +1,69 @@
+"""
+Implementation of "credit" XBlock service
+"""
+
+import logging
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
+
+#from opaque_keys.edx.keys import CourseKey
+
+from student.models import CourseEnrollment
+
+log = logging.getLogger(__name__)
+
+
+class CreditService(object):
+    """
+    Course Credit XBlock service
+    """
+
+    def get_credit_state(self, user_id, course_key):
+        """
+        Return all information about the user's credit state inside of a given
+        course.
+
+        ARGS:
+            - user_id: The PK of the User in question
+            - course_key: The course ID (as string)
+
+        RETURNS:
+            NONE (user not found or is not enrolled or is not credit course)
+            - or -
+            {
+                'enrollment_mode': the mode that the user is enrolled in the course
+                'credit_requirements': the requirements (dict) in order to eligible for credit
+                'credit_requirement_status': the user's status in fulfilling those requirements
+            }
+        """
+
+        # This seems to need to be here otherwise we get
+        # circular references when starting up the app
+        from openedx.core.djangoapps.credit.api.eligibility import (
+            is_credit_course,
+            get_credit_requirements,
+            get_credit_requirement_status,
+        )
+
+        # course_key = CourseKey.from_string(course_id)
+
+        try:
+            user = User.objects.get(id=user_id)
+        except ObjectDoesNotExist:
+            # bad user_id
+            return None
+
+        enrollment = CourseEnrollment.get_enrollment(user, course_key)
+        if not enrollment:
```

Comment:
It probably won't matter here, but it's possible for a student to have a course enrollment record if they enrolled and then unenrolled.  `CourseEnrollment.is_enrolled()` takes this into account by checking the `is_active` flag (as well as some CCX stuff that I don't understand).